www/api/class/centreon_monitoring_externalcmd.class.php

```diff
@@ -61,34 +61,78 @@ public function __construct()
             }
 
-            if (count($this->arguments['commands'])) {
-                if ($fh = @fopen($this->centcore_file, 'a+')) {
-                    foreach ($this->arguments['commands'] as $command) {
-                        if (isset($pollers[$command['poller_id']])) {
-                            fwrite($fh,
-                                "EXTERNALCMD:" . $command["poller_id"] . ":[" .
-                                $command['timestamp'] . "] " . $command['command'] . "\n"
+            $externalCommand = new CentreonExternalCommand();
+            $availableCommands = array();
+
+            /**
+             * We need to make the concordance between the data saved in the database
+             * and the action provided by the user.
+             */
+            foreach ($externalCommand->getExternalCommandList() as $key => $cmd) {
+                foreach ($cmd as $c) {
+                    $availableCommands[$c] = $key;
+                }
+            }
+
+            $isAdmin = $centreon->user->admin;
+
+            /**
+             * If user is not admin we need to retrieve its ACL
+             */
+            if (!$isAdmin) {
+                $userAcl = new CentreonACL($centreon->user->user_id, $centreon->centreon->admin);
+            }
+
+            if ($fh = @fopen($this->centcore_file, 'a+')) {
+                foreach ($this->arguments['commands'] as $command) {
+                    $commandSplitted = explode(';', $command['command']);
+                    $action = $commandSplitted[0];
+
+                    // checking that action provided exists
+                    if (!array_key_exists($action, $availableCommands)) {
+                        throw new RestBadRequestException('Action ' . $action . ' not supported');
+                    }
+
+                    if (!$centreon->user->admin) {
+                        // Checking that the user has rights to do the action provided
+                        if ($userAcl->checkAction($availableCommands[$action]) === 0) {
+                            throw new RestUnauthorizedException(
+                                'User is not allowed to execute ' . $action . ' action'
                             );
-                        } else {
-                            throw new RestException('Cannot open Centcore file');
                         }
                     }
-                    fclose($fh);
+
+                    if (isset($pollers[$command['poller_id']])) {
+                        fwrite($fh,
+                            "EXTERNALCMD:" . $command["poller_id"] . ":[" .
+                            $command['timestamp'] . "] " . $command['command'] . "\n"
+                        );
+                    }
                 }
+                fclose($fh);
+                return (array('success' => true));
+            } else {
+                throw new RestException('Cannot open Centcore file');
             }
-            return (array('success' => true));
```

Comment:
are you that this is not needed elsewhere ?

Reply:
it's ok cause `if` return this array, and `else` throw an exception